pkg/kubectl/cmd/patch.go

```diff
@@ -239,14 +233,26 @@ func RunPatch(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
 			return err
 		}
 
+		if !reflect.DeepEqual(info.Object, targetObj) {
+			dataChangedMsg = "patched"
+		}
+
 		// TODO: if we ever want to go generic, this allows a clean -o yaml without trying to print columns or anything
 		// rawExtension := &runtime.Unknown{
 		//	Raw: originalPatchedObjJS,
 		// }
-		if err := info.Refresh(targetObj, true); err != nil {
-			return err
+		if dataChangedMsg == "patched" {
```

Comment:
nit: rather than comparing strings here, just declare a bool  before the DeepEqual in the block above and check that here

Reply:
Thanks, good catch!

I moved the `didPatch` variable definition up one scope to allow reuse. Also created a tiny helper function to centralize the generation of the operation string.